apps/web/src/utils/logServerSideEvent.ts

```diff
@@ -0,0 +1,146 @@
+import crypto from 'crypto';
+import { logger } from 'apps/web/src/utils/logger';
+import { isDevelopment, analyticsConfig } from 'apps/web/src/constants';
+import { NextApiRequest } from 'apps/web/node_modules/next/dist/shared/lib/utils';
+
+type EventProperties = {
+  action: string;
+  context: string;
+  componentType: string;
+  projectName: string;
+  pagePath?: string;
+  error?: string;
+  platform?: string;
+  locale?: string | null;
+  sessionLccId?: string | null;
+  timeStart?: number;
+  pageKey?: string;
+  prevPageKey?: string;
+  prevPagePath?: string;
+  hasDoubleFired?: boolean;
+  sessionUuid?: string;
+  height?: number;
+  width?: number;
+  auth?: number;
+};
+type SupplementalEventData = {
+  userId?: string | null;
+  timestamp?: number;
+  eventId?: number;
+  sessionId?: number;
+  versionName?: string;
+  platform?: string;
+  osName?: string;
+  osVersion?: number;
+  deviceModel?: string;
+  language?: string;
+  userProperties?: unknown;
+  uuid?: string;
+  sequenceNumber?: number;
+  userAgent?: unknown;
+};
+
+type ServerSideEvent = {
+  eventType: string;
+  deviceId: string;
+  eventProperties: EventProperties;
+  timestamp: number;
+  sessionId: number;
+  platform: string;
+  userId?: string | null;
+  eventId?: number;
+  versionName?: string;
+  osName?: string;
+  osVersion?: number;
+  deviceModel?: string;
+  language?: string;
+  userProperties?: unknown;
+  uuid?: string;
+  sequenceNumber?: number;
+  userAgent?: unknown;
+};
+
+export default function logServerSideEvent(
+  eventName: string,
+  deviceId: string,
+  eventProperties: Omit<EventProperties, 'projectName'>,
+  supplementalEventData: SupplementalEventData = {},
+) {
+  const event = createEventData(eventName, deviceId, eventProperties, supplementalEventData);
+  const stringifiedEvent = JSON.stringify([convertKeys(event)]);
+  const uploadTime = new Date().getTime().toString();
+  const checksum = crypto
+    .createHash('md5')
+    .update(stringifiedEvent + uploadTime)
+    .digest('hex');
```

Comment:
nit: I don't think checksumming is really necessary

Reply:
I'm not 100% sure it's necessary, but the CCA network call definitely includes this for client-side log requests, so I think safer to leave in if okay with you